### PR TITLE
fix(auth): adds missing EMAIL_NOT_FOUND error code

### PR DESCRIPTION
**Pull request by bojeil-google**
Catch EMAIL_NOT_FOUND and translate to EmailNotFoundError  when /accounts:sendOobCode is called for password reset on a user that does not exist.

### CODE REVIEW

**Review by lahirumaramba — APPROVED** [review]
Thank you @bojeil-google !
LGTM. Please add a TW to review the reference doc changes.

**Review by hiranya911 — CHANGES_REQUESTED** [review]
One possible typo. Can we also add a test case? Similar to: https://github.com/firebase/firebase-admin-python/blob/ebf09619ad24cdcf04a8a81362844233fa2a2130/tests/test_user_mgt.py#L286

File: firebase_admin/_auth_client.py [hunk 1]
```diff
@@ -464,6 +465,7 @@ def generate_email_verification_link(self, email, action_code_settings=None):
 
         Raises:
             ValueError: If the provided arguments are invalid
+            UserNotFoundError: If no user exists by the specified email address.
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply firebase_admin/_auth_client.py:468]: Unfortunately, this is the problem. They throw different errors. For password reset, they throw EMAIL_NOT_FOUND. For email verification, they throw USER_NOT_FOUND.

**Comment by bojeil-google** [comment]
> One possible typo. Can we also add a test case? Similar to:
> 
> https://github.com/firebase/firebase-admin-python/blob/ebf09619ad24cdcf04a8a81362844233fa2a2130/tests/test_user_mgt.py#L286

Done

File: firebase_admin/_auth_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hiranya911) [reply firebase_admin/_auth_client.py:468]: Ah, fair enough 👍

**Review by hiranya911 — APPROVED** [review]
Thanks. LGTM!

File: firebase_admin/_auth_client.py [hunk 2]
```diff
@@ -444,6 +444,7 @@ def generate_password_reset_link(self, email, action_code_settings=None):
 
         Raises:
             ValueError: If the provided arguments are invalid
+            EmailNotFoundError: If no user exists by the specified email address.
```
Comment (egilmorez) [inline firebase_admin/_auth_client.py:447]: Is there a specific reason to use "by" here?

I'd expect "for" probably, here and below.

**Review by egilmorez — APPROVED** [review]
LG, one tiny nit.

File: firebase_admin/_auth_client.py (hunk 2 above)
Comment (bojeil-google) [reply firebase_admin/_auth_client.py]: Thanks for catching. I copied and pasted blindly. It is fixed now here and elsewhere.

**Pull request merged**